tasks/main.yml

```diff
@@ -49,6 +54,32 @@
     - mssql_upgrade | bool
     - mssql_version | int == 2017
 
+- name: Verify if the mssql_ha_replica_type variable is provided correctly
+  assert:
+    that:
+      - mssql_ha_replica_type in
+          ['primary', 'synchronous', 'witness', 'absent']
+    fail_msg: >-
+      You must set the mssql_ha_replica_type variable to one of 'primary',
+      'synchronous', 'witness', 'absent'
+  when: mssql_ha_configure | bool
+
+- name: Verify that 'mssql_ha_replica_type = primary' is provided once
+  assert:
+    that: |-
+      {% set result = namespace(total=0) %}
+      {% for item in ansible_play_hosts_all %}
+      {% if hostvars[item]['mssql_ha_replica_type'] == 'primary' %}
+      {% set result.total = 1 + result.total %}
+      {% endif %}
+      {% endfor %}
+      {{ result.total }} == 1
```

Comment:
In almost every case, doing something like `x = x + 1` in a loop in ansible or jinja is an anti-pattern.  I think you can do something like this instead:
```yaml
- name: Verify that 'mssql_ha_replica_type = primary' is provided once
  assert:
    that: ansible_play_hosts_all | map('extract', hostvars, 'mssql_ha_replica_type') | select('match', '^primary$') | length == 1
```
That is, ensure that at least one, and only one, host has `mssql_ha_replica_type=primary`

Reply:
Done, thank you, your solution works